src/components/layouts/pages/inspect/tree/Tree.js

```diff
@@ -0,0 +1,165 @@
+import React, { useState, Fragment } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { getRuleSummaries } from '../../../../../store/job/result/selectors';
+import { getJobProfile } from '../../../../../store/job/selectors';
+import List from '@material-ui/core/List';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import Collapse from '@material-ui/core/Collapse';
+import Tooltip from '@material-ui/core/Tooltip';
+import Chip from '@material-ui/core/Chip';
+import ExpandLess from '@material-ui/icons/ExpandLess';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+import './Tree.scss';
+import errorMap from '../validationErrorMessages_en.json';
+
+const LIST_HEADER = 'Errors overview';
+const UNSELECTED = -1;
+
+function Tree({ ruleSummaries, profile }) {
+    const [openedRule, setOpenedRule] = useState(UNSELECTED);
+    const [selectedCheck, setSelectedCheck] = useState(UNSELECTED);
+    const onRuleClick = index => {
+        if (openedRule === index) {
+            return setOpenedRule(UNSELECTED);
+        }
+
+        return setOpenedRule(index);
+    };
+    const onCheckClick = context => setSelectedCheck(context);
+
+    return (
+        <section className="summary-tree">
+            <List
```

Comment:
Refactoring, do as a separate commit as a last resort, if no time this will be postponed to M2.0:

Let's move generic accordion behavior as a separate component with the props like below (it's not exact proposal, just an idea, so you can adapt it a bit during refactoring):
```
items: [{
    label: "",
    actions: [{
        icon: iconElement,
        onClick:  //handler to open modal will be passed here
    }],
    children: [{
        label: "",
        onClick: // handler to select check will be passed here
    }],
}]
```

Accordion should encapsulate collapsing logic and display dynamic action buttons. How to react on those actions should be defined outside Accordion itself.

Reply:
Will be done in scope of #18